ocaml/tests/test_nm.ml

```diff
@@ -0,0 +1,72 @@
+open Test_highlevel
+open Test_common
+open OUnit
+
+module MaybeUpdateMasterPifMac = Generic.Make(Generic.EncapsulateState(struct
+  module Io = struct
+    type input_t = (bool * string option) (* auto_update_mac, new_primary_slave_mac *)
+
+    type output_t = (string * string)  (* record bond master mac, db bond master mac *)
+
+    let show_auto_update_mac = fun auto_update_mac -> "auto_update_mac="^string_of_bool auto_update_mac
```

Comment:
The `Test_printers` module has handy combinators for this kind of thing; `bool` for this case.